### PR TITLE
georgettica/app interface old way

### PR DESCRIPTION
**Pull request by georgettica**
- chore(appsre): copy pasta all
- chore(appsre): sed names
- feat(bundle): create bundle

### CODE REVIEW

**Comment by georgettica** [comment]
removed WIP as I want others to 👀  and give me the 👍  that is should work

File: bundle/manifests/route-monitor-operator.clusterserviceversion.yaml
Comment (RiRa12621) [inline bundle/manifests/route-monitor-operator.clusterserviceversion.yaml:42]:
```suggestion
    url: https://github.com/openshift/route-monitor-operator/
```

File: config/manifests/bases/route-monitor-operator.clusterserviceversion.yaml
Comment (RiRa12621) [inline config/manifests/bases/route-monitor-operator.clusterserviceversion.yaml:53]:
```suggestion
    url: https://github.com/openshift/route-monitor-operator/
```

**Review by RiRa12621 — APPROVED** [review]
/lgtm
just small suggestion to make the github repo the home page

additionally we may want to add a test in the makefile that runs the python script but that should not and does not have to be part of this PR

**Comment by RiRa12621** [comment]
/assign @bdematte @rporres 
unhold when you're ready

**Comment by georgettica** [comment]
thanks @RiRa12621 !

**Comment by RiRa12621** [comment]
@georgettica ping me when you're fully done for LGTM again

**Comment by georgettica** [comment]
For me I think it's done. But still have some things that we need to solve for the short/long term.
As discussed privately, @rporres knows the 🛣️.

But from.what I can recall, we were thinking of clearing most of if not all of `hack/generate-operator-bundle.py`

What is missing is being able to manually set the `replaces` field

File: Makefile
Comment (rporres) [inline Makefile:8]: This is only used once in `148`. Why not using there `OPERATOR_VERSION` directly?

File: hack/app_sre_create_image_catalog.sh
Comment (rporres) [inline hack/app_sre_create_image_catalog.sh:30]: Please make sure this file is going exist in app-interface or the logic won't work 😅

File: hack/app_sre_create_image_catalog.sh
Comment (rporres) [inline hack/app_sre_create_image_catalog.sh:56]: @georgettica and I worked yesterday in an approach that didn't need this script anymore: using `make bundle` and copying the `bundle/manifests` file

File: hack/app_sre_pr_check.sh
Comment (rporres) [inline hack/app_sre_pr_check.sh:14]: copy/pasta :)

File: hack/generate-operator-bundle.py
Comment (rporres) [inline hack/generate-operator-bundle.py:1]: this file should not be needed anymore

File: hack/validate_yaml.py
Comment (rporres) [inline hack/validate_yaml.py:1]: Where is this file used?

File: hack/app_sre_pr_check.sh
Comment (rporres) [inline hack/app_sre_pr_check.sh:1]: Is this file going to be used at all? I understand this operator will run PR checks via Prow

**Comment by rporres** [comment]
> What is missing is being able to manually set the replaces field

This cannot be merged without that

**Comment by bdematte** [comment]
> > What is missing is being able to manually set the replaces field
> 
> This cannot be merged without that

I am not sure what this refers to, as the script is setting the `replaces` field (with the value provided by the caller) so looks fine to me

**Comment by rporres** [comment]
> I am not sure what this refers to, as the script is setting the replaces field (with the value provided by the caller) so looks fine to me

In our yesterday's tests, replaces field was being set to something like `0.<current_minor - 1>.<current_patch>-<current_sha>`

**Comment by georgettica** [comment]
what @rporres means, is if you omit fully the python file, the operator on it's own set the replaces field. we can join the two together where the python script will only change the replaces field according to it's liking.

hope that clears the issue @bdematte 

also if you would like, I can show you the contents of `make generate` and what the downstream repo look like and I think you will see the divergance

File: hack/app_sre_pr_check.sh
Comment (georgettica) [reply hack/app_sre_pr_check.sh:1]: as of now the operator in app-sre is pointing to this script.

if it's in prow that's great! It's better to be tested twice then none at all.

also IIUC this is the entrypoint if the make target doesn't exist, so the file is required

File: hack/validate_yaml.py
Comment (georgettica) [reply hack/validate_yaml.py:1]: IDK, ran `cp -R aws-account-operator/hack/. route-monitor-operator/hack` and kept the results, if it's not used I see no reason to keep it

File: hack/app_sre_pr_check.sh
Comment (georgettica) [reply hack/app_sre_pr_check.sh:14]: missed one :(

File: hack/app_sre_create_image_catalog.sh
Comment (georgettica) [reply hack/app_sre_create_image_catalog.sh:30]: I can test this code path with you if you wish

File: Makefile
Comment (georgettica) [reply Makefile:8]: I wanted to keep the codebase (which was copied from operator-sdk) the same as much as I can. if we really want to change it I can live with that

File: hack/generate-operator-bundle.py
Comment (georgettica) [reply hack/generate-operator-bundle.py:1]: As I said multiple times, not 100% true

File: hack/app_sre_create_image_catalog.sh
Comment (georgettica) [reply hack/app_sre_create_image_catalog.sh:56]: not exactly, as you mentioned we need the `replaces` filed to be calculated as we want. so we can just override in python for now

File: hack/app_sre_pr_check.sh
Comment (georgettica) [reply hack/app_sre_pr_check.sh:1]: oh, it's PR check.. did'nt read it good the first time.. will delete

**Comment by georgettica** [comment]
@rporres @bdematte I am ready for another round of reviews

File: hack/app_sre_create_image_catalog.sh
Comment (rporres) [reply hack/app_sre_create_image_catalog.sh:30]: Just make sure it exists whenever you create the saas file in app-interface

File: hack/app_sre_create_image_catalog.sh
Comment (rporres) [reply hack/app_sre_create_image_catalog.sh:56]: I understood you were working in the manifest generation using `make bundle`. Hence my comment

File: hack/generate-operator-bundle.py
Comment (rporres) [reply hack/generate-operator-bundle.py:1]: I don't really understand why we were working in the `make bundle` approach then. But if you want to keep this and work on that later, I'm fine with it

File: hack/validate_yaml.py
Comment (rporres) [reply hack/validate_yaml.py]: I would not keep it if it's not in use

**Review by rporres — APPROVED** [review]
ok, so we now use the python code only to deal with the csv... not ideal but looks better. I guess it should be changed in the future but if it works for now, let's unblock this

**Comment by georgettica** [comment]
@RiRa12621 it seems we have the green light for now, so:
/unhold

**Comment by georgettica** [comment]
for future reference, the commands that can be used are:
```
kustomize build config/manifests/ | operator-sdk generate packagemanifests --output-dir saas-route-monitor-operator-bundle/route-monitor-operator/ --version $FIRST_VERSION
kustomize build config/manifests/ | operator-sdk generate packagemanifests --output-dir saas-route-monitor-operator-bundle/route-monitor-operator/ --from-version  $FIRST_VERSION --version $SECOND_VERSION
```

**Comment by georgettica** [comment]
if @rporres approved, then it's:
/lgtm ?
or should someone else (@RiRa12621 ) do it

**Pull request merged**
